### PR TITLE
Add VLR test to internal wave case

### PR DESCRIPTION
**Pull request by cbegeman**
Currently, there are no tests of the vertical Lagrangian remapping capability in compass. This PR adds a VLR version for the default and RPE tests of the internal wave test group.

Checklist
* [X] User's Guide has been updated
* [X] Developer's Guide has been updated
* [X] Documentation has been [built locally](https://mpas-dev.github.io/compass/latest/developers_guide/building_docs.html) and changes look as expected
* [X] Document (in a comment titled `Testing` in this PR) any testing that was used to verify the changes
* [X] New tests have been added to a test suite

### CODE REVIEW

**Comment by cbegeman** [comment]
### Testing

The default and rpe_test with VLR off and VLR on have been run on compy with intel, impi compiler. Results appear as expected.

File: compass/ocean/tests/internal_wave/default/__init__.py [hunk 1]
```diff
@@ -10,18 +10,26 @@ class Default(TestCase):
...
         ----------
         test_group : compass.ocean.tests.internal_wave.InternalWave
             The test group that this test case belongs to
+        vlr : boolean
+            Whether vertical Lagrangian remapping will be tested
         """
-        super().__init__(test_group=test_group, name='default')
+        name = 'default'
+        if vlr:
+            subdir = 'vlr/' + name
```
Comment (xylar) [inline compass/ocean/tests/internal_wave/default/__init__.py:26]: I realize it works but I prefer that we not use string addition.
```suggestion
            subdir = f'vlr/{name}' 
```

File: compass/ocean/tests/internal_wave/default/__init__.py [hunk 2]
```diff
@@ -10,18 +10,26 @@ class Default(TestCase):
...
-    def __init__(self, test_group):
+    def __init__(self, test_group, vlr=False):
         """
         Create the test case
 
         Parameters
         ----------
         test_group : compass.ocean.tests.internal_wave.InternalWave
             The test group that this test case belongs to
+        vlr : boolean
```
Comment (xylar) [inline compass/ocean/tests/internal_wave/default/__init__.py:21]: Optional arguments (keyword arguments) should always have `, optional` in the associated part of the docstring.
```suggestion
        vlr : boolean, optional
```

File: compass/ocean/tests/internal_wave/forward.py [hunk 3]
```diff
@@ -8,7 +8,7 @@ class Forward(Step):
     test cases.
     """
     def __init__(self, test_case, name='forward', subdir=None, ntasks=1,
-                 min_tasks=None, openmp_threads=1, nu=None):
+                 min_tasks=None, openmp_threads=1, nu=None, vlr=False):
```
Comment (xylar) [inline compass/ocean/tests/internal_wave/forward.py:11]: This needs to be added to the docstring.

File: compass/ocean/tests/internal_wave/forward.py [hunk 4]
```diff
@@ -50,6 +50,11 @@ def __init__(self, test_case, name='forward', subdir=None, ntasks=1,
             options = {'config_mom_del2': '{}'.format(nu)}
             self.add_namelist_options(options)
 
+        if vlr:
+            # turn vertical Lagrangian-remapping on
+            self.add_namelist_options({
+                'config_vert_advection_method': '"remap"'})
```
Comment (xylar) [inline compass/ocean/tests/internal_wave/forward.py:56]: I could be wrong but I think we always use single-quotes in namelist files.
```suggestion
                'config_vert_advection_method': "'remap'"})
```

Comment (xylar) [reply compass/ocean/tests/internal_wave/forward.py:56]: Thankfully, this issue goes away with the yaml files in polaris.

File: compass/ocean/tests/internal_wave/rpe_test/__init__.py [hunk 5]
```diff
@@ -11,7 +11,7 @@ class RpeTest(TestCase):
     5 different values of the viscosity at the given resolution.
     """
 
-    def __init__(self, test_group):
+    def __init__(self, test_group, vlr=False):
```
Comment (xylar) [inline compass/ocean/tests/internal_wave/rpe_test/__init__.py:14]: Again, needs to be added to the docstring.

**Review by xylar — CHANGES_REQUESTED** [review]
I have a few formatting and documentation related changes I'd like to see.  But I tested the `pr` suite on Chrysalis with these tests included and they passed just fine.  (The suite is still running but I'll update this comment as soon as it finishes).  They also add only 2 seconds to the total runtime!

**Comment by xylar** [comment]
@cbegeman, I don't know if this is something you have time to return to but it seems like it would be quick to make these formatting changes and get this merged in.

File: compass/ocean/tests/internal_wave/rpe_test/__init__.py (hunk 5 above)
Comment (cbegeman) [reply compass/ocean/tests/internal_wave/rpe_test/__init__.py:14]: Addressed by https://github.com/MPAS-Dev/compass/pull/564/commits/74755efd7497e3937d080aeea63ce0400564cf4d

File: compass/ocean/tests/internal_wave/forward.py (hunk 3 above)
Comment (cbegeman) [reply compass/ocean/tests/internal_wave/forward.py:11]: Addressed by https://github.com/MPAS-Dev/compass/pull/564/commits/74755efd7497e3937d080aeea63ce0400564cf4d

**Comment by cbegeman** [comment]
@xylar Thanks for the reminder. I reran the default vlr case with these changes and it was successful.

File: compass/ocean/tests/internal_wave/default/__init__.py [hunk 6]
```diff
@@ -10,18 +10,26 @@ class Default(TestCase):
...
+    def __init__(self, test_group, vlr=False):
         """
         Create the test case
 
         Parameters
         ----------
         test_group : compass.ocean.tests.internal_wave.InternalWave
             The test group that this test case belongs to
+        vlr : boolean, optional
+            Whether vertical Lagrangian remapping will be tested
```
Comment (xylar) [inline compass/ocean/tests/internal_wave/default/__init__.py:22]:
```suggestion
        test_group : compass.ocean.tests.internal_wave.InternalWave
            The test group that this test case belongs to

        vlr : bool, optional
            Whether vertical Lagrangian remapping will be tested
```

File: compass/ocean/tests/internal_wave/rpe_test/__init__.py [hunk 7]
```diff
@@ -11,18 +11,25 @@ class RpeTest(TestCase):
...
         """
         Create the test case
 
         Parameters
         ----------
         test_group : compass.ocean.tests.internal_wave.InternalWave
             The test group that this test case belongs to
+
+        vlr : boolean, optional
```
Comment (xylar) [inline compass/ocean/tests/internal_wave/rpe_test/__init__.py:23]:
```suggestion
        vlr : bool, optional
```

**Review by xylar — APPROVED** [review]
Two more tiny formatting fixes to docstrings if you don't mind, and I'll merge.

**Comment by xylar** [comment]
I'm going ahead with the 2 tiny changes I suggested yesterday so this can get merged.

**Comment by cbegeman** [comment]
@xylar Thanks for the detailed review!

**Pull request merged**